src/main/java/dao/Dao.java

```diff
@@ -0,0 +1,18 @@
+package dao;
+
+import java.util.List;
+
+public interface Dao<T> {
+
+    T get(int id);
+
+    T get(String name);
+
+    List<T> getAll();
+
+    void save(T t);
+
+    void update(T t, String[] params);
```

Comment:
why array?

Reply:
it was default example, for now I deleted this method as we do not need it now. As soon as we need it, will add.